### PR TITLE
feat(web): Hub spot identify

### PR DESCRIPTION
**Pull request by scopsy**
### What changed? Why was the change needed?

- On signup this properly ties the Hubspot entity with the new account.
- 
### Screenshots

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Special notes for your reviewer

</details>

### CODE REVIEW

File: apps/api/src/app/analytics/analytics.controller.ts [hunk 1]
```diff
@@ -26,4 +31,41 @@ export class AnalyticsController {
...
     };
   }
+
+  @Post('/identify')
+  @ExternalApiAccessible()
+  @UserAuthentication()
+  async identifyUser(@Body() userData: any, @UserSession() user: UserSessionData): Promise<any> {
+    try {
+      const hubspotPortalId = '44416662';
+      const hubspotFormId = 'fc39aa98-4285-4322-9514-52da978baae8';
```
Comment (scopsy) [inline apps/api/src/app/analytics/analytics.controller.ts]: Left them here hardcoded, as those are public values.

File: apps/web/src/api/telemtry.ts
Comment (djabarovgeorge) [inline apps/web/src/api/telemtry.ts:1]: can we rename the file to `telemetry` instead?

File: apps/web/src/api/api.client.ts [hunk 2]
```diff
@@ -45,7 +45,7 @@ export const api = {
   },
   post(url: string, payload, params?: CustomDataType) {
     return axios
-      .post(`${API_ROOT}${url}`, payload, { params, headers: getHeaders() })
+      .post(`${API_ROOT}${url}`, payload, { params, headers: { ...getHeaders() } })
```
Comment (djabarovgeorge) [inline apps/web/src/api/api.client.ts]: what is the difference here between `getHeaders()` and `{ ...getHeaders() }`?

File: apps/api/src/app/analytics/analytics.controller.ts (hunk 1 above)
Comment (djabarovgeorge) [reply apps/api/src/app/analytics/analytics.controller.ts]: Do those identifiers tend to change? if there are, should we consider storing them in the environment variables?

File: apps/api/src/app/analytics/analytics.controller.ts
Comment (djabarovgeorge) [inline apps/api/src/app/analytics/analytics.controller.ts:1]: All of this logic could be a use case :), for your consideration of the refactoring is needed

**Review by djabarovgeorge — APPROVED** [review]
go go analytics!!! awesome work I added some comments to your review

File: apps/web/src/ee/clerk/components/QuestionnaireForm.tsx [hunk 3]
```diff
@@ -58,6 +59,20 @@ export function QuestionnaireForm() {
     };
     await updateOrganizationMutation(updateClerkOrgDto);
 
+    const hubspotContext = document.cookie
```
Comment (SokratisVidros) [inline apps/web/src/ee/clerk/components/QuestionnaireForm.tsx]: You can use [js-cookies](https://github.com/novuhq/novu/blob/5cd0e5a6a1c9aaa0fd042b62959d78ae02c58994/apps/web/package.json#L99) to get the hubspot cookie value

**Review by SokratisVidros — COMMENTED** [review]
Does this change ensure that Hubspot context is the same across subdomains, for example between nocu.co and dashboard.novu.co?

File: apps/web/src/api/api.client.ts (hunk 2 above)
Comment (scopsy) [reply apps/web/src/api/api.client.ts]: Good catch, I just had to pass some headers here previously. Not needed anymore. Reverted.

**Comment by scopsy** [comment]
Yes 🙏 Did a few tests and so far so good. Will also monitor this in the wild

**Pull request merged**